src/Router.sol

```diff
@@ -105,7 +107,7 @@ contract Router is IRouter, EIP712, FeeGenerator {
     /// @param user The delegator of the sender
     /// @return Valid or not
     function isValidDelegateeFor(address user) public view returns (bool) {
-        return block.timestamp <= delegatees[user][msg.sender];
+        return block.timestamp <= uint256(delegations[user][msg.sender].expiry);
```

Comment:
I'm wondering if it's a good idea to use `uint128` for expiry. Eventually, the block timestamp will be over uint128.max and we will have no valid delegatee.

Reply:
In practice it's not quite possible for timestamp to exceed `uint128`. [Permit2](https://github.com/Uniswap/permit2/blob/main/src/interfaces/IAllowanceTransfer.sol#L83C1-L83C1) library uses `uint48` to store expiration.